pkg/web/api/v1/transaction.go

```diff
@@ -221,10 +221,94 @@ func (c *Transaction) Model() (model *models.Transaction, err error) {
 	return model, nil
 }
 
+//===========================================================================
+// Transaction Status Helpers
+//===========================================================================
+
+const (
+	color   = "color"
+	tooltip = "tooltip"
+)
+
 func (c *Transaction) TitleStatus() string {
 	return cases.Title(language.English).String(c.Status)
 }
 
+func (c *Transaction) ColorStatus() string {
+	return getStatus(c.Status, color)
+}
+
+func (c *Transaction) TooltipStatus() string {
+	return getStatus(c.Status, tooltip)
+}
+
+func getStatus(status, attr string) string {
```

Comment:
I like the idea here to unify handling in a single `getStatus` method; having the color and the tool tip in the same function just a few lines apart makes it easier to edit them together. In a compiled language like Go, however, the nested if statements inside of the switch will prevent any compiler optimization from occurring. This is one of the reasons Go is so verbose; because we're trying to make maximal use of the CPU and be as fast as possible, the code allows us to to take advantage of the compiler optimizations now and in the future. 

So a function like `getStatus` is better in Python, Ruby, R, and Javascript than it is in C, Go, Rust, or Java. 

I'll submit a commit that does the refactoring and uses constants to do what you intended; make the editing of the statuses a bit easier; it's just that there will be two switch statements instead of one.

Reply:
I initially had two separate switch statements, but then thought it felt redundant; so I really appreciate this explanation.